src/Database/TigerBeetle/Internal/FFI/Client.hsc

```diff
@@ -0,0 +1,36 @@
+{-# LANGUAGE ForeignFunctionInterface #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE OverloadedRecordDot #-}
+{-# LANGUAGE StrictData #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE RecordWildCards #-}
+
+module Database.TigerBeetle.Internal.FFI.Client where
+
+import Data.Word
+import Foreign.Ptr
+import Foreign.Storable
+import Data.Vector (Vector)
+import Data.Vector qualified as V
+
+#include "tb_client.h"
+
+data TBClient = TBClient
+    { tbClientOpaque :: Vector Word64
+    }
+    deriving (Show, Eq)
+
+instance Storable TBClient where
+    sizeOf _ = #{size tb_client_t}
+
+    alignment _ = #{alignment tb_client_t}
+
+    peek ptr = do
+      let opaquePtr = #{ptr tb_client_t, opaque} ptr
+      tbClientOpaque <- V.generateM 4 (\i -> peekByteOff opaquePtr (i * 8))
```

Comment:
note: I think once we get everything working we could potentially look at `Data.Array` here instead of `Vector` where we have bounded data, if that makes sense.